app/src/main/java/com/stonks/ui/currency/CurrencyFragment.kt

```diff
@@ -0,0 +1,193 @@
+import kotlinx.android.synthetic.main.fragment_currency.*
+
+class CurrencyFragment : Fragment() {
+    private val TAG = CurrencyFragment::class.java.name
+    private var disposable: Disposable? = null
+    private lateinit var baseRateSpinnerString: String
+    private lateinit var targetRateSpinnerString: String
+    private var isNumeric: Boolean = false
+
+    override fun onCreateView(
+            inflater: LayoutInflater, container: ViewGroup?,
+            savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_currency, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val ratesNameArray = resources.getStringArray(R.array.rates)
+        val chartPrimaryRatesArray = resources.getStringArray(R.array.primary_rates)
+
+        val currencyFragmentUtils = CurrencyFragmentUtils(disposable = disposable)
+        val currencyLineChart = StockLineChart(currency_chart)
+
+        initPrimaryRatesName(chartPrimaryRatesArray)
+
+        currencyFragmentUtils.setLastUpdatedDate(last_date_update, "Data for: ")
+
+        base_rate_spinner?.setSelection(DEFAULT_CURRENCY_INDEX)
+        base_rate_spinner?.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(
+                    parent: AdapterView<*>?,
+                    view: View?,
+                    position: Int,
+                    id: Long
+            ) {
+                baseRateSpinnerString = ratesNameArray[position].split(",")[0]
+                rate_number.setText("1.0")
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>?) {}
+        }
+
+        target_rate_spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(
+                    parent: AdapterView<*>?,
+                    view: View?,
+                    position: Int,
+                    id: Long
+            ) {
+                targetRateSpinnerString = ratesNameArray[position].split(",")[0]
+                rate_number.setText("1.0")
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>?) {}
+        }
+
+        currency_button_group.addOnButtonCheckedListener { group, checkedId, isChecked ->
+            if (isChecked) {
+                when (currency_button_group.checkedButtonId) {
+                    R.id.currency_togglebutton_one_week_selector -> {
+                        currencyFragmentUtils.plotRatesPerPeriod(
+                                startDate = "1W",
+                                targetRate = targetRateSpinnerString,
+                                baseRate = baseRateSpinnerString,
+                                stockLineChart = currencyLineChart,
+                                currencyChart = currency_chart
+                        )
```

Comment:
Add 1D button back. We MUST have it according to TZ. We just can say, that we have no data for the period selected (like a toast, or any other way), but we MUST provide an opportunity to select 1 day

Reply:
Added